jose/jws.py

```diff
@@ -214,12 +220,14 @@ def _verify_signature(signing_input, header, signature, key='', algorithms=None)
         if algorithms is not None and alg not in algorithms:
             raise JWSError('The specified alg value is not allowed')
 
-        try:
-            key = jwk.construct(key, alg)
+        if 'keys' in key:  # JWK Set per RFC 7517
+            keys = key['keys']
+        else:
+            keys = [key]
 
-            if not key.verify(signing_input, signature):
+        try:
+            if not _sig_matches_keys(keys, signing_input, signature, alg):
                 raise JWSSignatureError()
-
         except JWSSignatureError:
```

Comment:
I was hesitant to change this, but why were you catching `JWSSignatureError` here, and then re-raising a generic `JWSError` with text explaining it's a signature failure? Since `JWSSignatureError` is already a subclass of `JWSError`, why not just let the `JWSSignatureError` propagate up?